src/tree.rs

```diff
@@ -1147,4 +1270,34 @@ mod tree_tests {
         let sub_root = tree.find_subtree_root_between_ids(&root_id, &node_4_id); //invert for None
         assert_eq!(sub_root, None);
     }
+
+    #[test]
+    fn test_swap_sub_trees() {
+        let mut tree = Tree::new();
```

Comment:
I like these tests very much, but I feel like there is another case to test.  Can't think of anything off the top of my head though, so if you think this is exhaustive I'm fine with it.